PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -148,6 +148,31 @@ def miniAOD_customizeCommon(process):
     process.patJets.userData.userFunctionLabels = cms.vstring('vtxMass','vtxNtracks','vtx3DVal','vtx3DSig','vtxPx','vtxPy','vtxPz','vtxPosX','vtxPosY','vtxPosZ')
     process.patJets.tagInfoSources = cms.VInputTag(cms.InputTag("pfSecondaryVertexTagInfos"))
     process.patJets.addTagInfos = cms.bool(True)
+
+    ## HIP mitigation for b tagging
+    ## (this will run below-specified taggers with the HIP mitigation enabled
+    ## and will update 'selectedPatJets' by attaching to them extra b-tag discriminators
+    ## which will be stored with the 'hip' prefix added to the usual discriminator names)
+    from PhysicsTools.PatAlgos.tools.jetTools import updateJetCollection
+    updateJetCollection(
+        process,
+        jetSource = cms.InputTag('selectedPatJets'),
+        ## updateJetCollection defaults to MiniAOD inputs. Here, this needs to be changed to RECO/AOD inputs
+        pvSource = cms.InputTag('offlinePrimaryVertices'),
+        pfCandidates = cms.InputTag('particleFlow'),
+        svSource = cms.InputTag('inclusiveCandidateSecondaryVertices'),
+        muSource = cms.InputTag('muons'),
+        elSource = cms.InputTag('gedGsfElectrons'),
+        ##
+        jetCorrections = ('AK4PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], ''),
+        btagDiscriminators = ["pfJetBProbabilityBJetTags", "pfJetProbabilityBJetTags",
+                              "pfCombinedSecondaryVertexV2BJetTags", "pfCombinedInclusiveSecondaryVertexV2BJetTags",
+                              "softPFMuonBJetTags", "softPFElectronBJetTags", "pfCombinedMVAV2BJetTags"],
```

Comment:
If I'm not mistaken these entries trigger rerun of softPFMuonBJetTags and softPFElectronBJetTags
(both the hipsoftPFMuonsTagInfosHIP and hipsoftPFMuonBJetTagsHIP are called).
Is it justifiable? Settings are unchanged for the soft lepton taggers (jets change to updatedPatJetsHIP, but these are technically the same jets).

Is it because of the pfCombined that needs the soft leptons and can't pick up values from softPF{Muon,Electron}BJetTags?